src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -215,10 +215,10 @@ internal Binder GetSpeculativeBinder(int position, ExpressionSyntax expression,
             }
 
             // May be binding an expression in a context that doesn't have a LocalScopeBinder in the chain.
-            return new LocalScopeBinder(binder);
+            return new ExpressionVariableBinder(expression, binder);
```

Comment:
>ExpressionVariableBinder [](start = 23, length = 24)

I think the code will be more robust long term if we will not create ExpressionVariableBinder explicitly, but instead always create an ExecutableCodeBinder and then request a binder for the root from it.